src/sources/eventstoredb_metrics/mod.rs

```diff
@@ -0,0 +1,162 @@
+        "eventstoredb_metrics"
+    }
+}
+
+fn eventstoredb(
+    endpoint: &str,
+    interval: u64,
+    namespace: Option<String>,
+    cx: SourceContext,
+) -> crate::Result<super::Source> {
+    let mut out = cx
+        .out
+        .sink_map_err(|error| error!(message = "Error sending metric.", %error));
+    let mut ticks = IntervalStream::new(tokio::time::interval(Duration::from_secs(interval)))
+        .take_until(cx.shutdown);
+    let tls_settings = TlsSettings::from_options(&None)?;
+    let client = HttpClient::new(tls_settings)?;
+    let url: Uri = format!("{}/stats", endpoint).parse()?;
+
+    Ok(Box::pin(
+        async move {
+            while ticks.next().await.is_some() {
+                let req = Request::get(&url)
+                    .header("content-type", "application/json")
+                    .body(Body::empty())
+                    .unwrap();
+
+                match client.send(req).await {
+                    Err(error) => {
+                        emit!(EventStoreDbMetricsHttpError {
+                            error: error.into(),
+                        });
+                        continue;
+                    }
+
+                    Ok(resp) => {
+                        let bytes = match hyper::body::to_bytes(resp.into_body()).await {
+                            Ok(b) => b,
+                            Err(error) => {
+                                emit!(EventStoreDbMetricsHttpError {
+                                    error: error.into(),
+                                });
+                                continue;
+                            }
+                        };
+
+                        match serde_json::from_slice::<Stats>(bytes.as_ref()) {
+                            Err(error) => {
+                                emit!(EventStoreDbStatsParsingError { error });
+                                continue;
+                            }
+
+                            Ok(stats) => {
+                                let metrics = stats.metrics(namespace.clone());
+
+                                emit!(EventStoreDbMetricsReceived {
+                                    events: metrics.len(),
+                                    byte_size: bytes.len(),
+                                });
+
+                                let mut metrics = stream::iter(metrics).map(Event::Metric).map(Ok);
+                                if out.send_all(&mut metrics).await.is_err() {
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        .map(Ok)
+        .boxed(),
+    ))
+}
+
+#[cfg(all(test, feature = "eventstoredb_metrics-integration-tests"))]
+mod integration_tests {
+    use super::*;
+    use crate::{test_util, Pipeline};
+    use tokio::time::Duration;
+
+    const EVENTSTOREDB_SCRAP_ADDRESS: &str = "http://localhost:2113/";
```

Comment:
@ktff Are you sure when starting your EventStoreDB service that is available on `localhost` ?

Reply:
@YoEight tests can be run with command
```
make test-integration-eventstoredb_metrics
```

This will start the `EventStoreDB`, run the tests, and then shutdown the service.

The command that starts the service is 
https://github.com/timberio/vector/blob/0c9111f695a14e5ca581e765f409211cd6064f11/scripts/setup_integration/eventstoredb_metrics_integration_env.sh#L22-L24

this will pull the image `eventstore/eventstore` and run it in docker/podman with arguments ` --insecure --stats-period-sec=1 `.

> Are you sure when starting your EventStoreDB service that is available on localhost ?

Yes, since it does get a response, but it's empty.